adr/022-transformation-engine.md

```diff
@@ -0,0 +1,28 @@
+# 22. Transformation Engine
+
+Date: 2024-06-11
+
+## Decision
+
+As with the [Validation Rule Engine](https://github.com/CDCgov/trusted-intermediary/blob/f11891504f5303657cfe1b5c61a515013265f239/adr/021-validation-engine.md), we will use a variation of the [Rules Engine Pattern](https://deviq.com/design-patterns/rules-engine-pattern) to design and implement a transformation rule engine that can transform incoming FHIR messages given a condition. Each transformation will be represented as a rule, specifying conditions that must be met and the corresponding failure message if the transformation fails. Initially, rules will be stored in a JSON file, with considerations on transitioning to a database for scalability.
```

Comment:
@basiliskus I updated this whole document based on the Validation Rules Engine ADR so I'm assuming it will need corrections/updates